src/java.base/share/classes/java/util/zip/ZipEntry.java

```diff
@@ -91,20 +91,23 @@ public class ZipEntry implements ZipConstants, Cloneable {
      */
     private static final long UPPER_DOSTIME_BOUND =
             128L * 365 * 24 * 60 * 60 * 1000;
-
+    // CEN header size + name length + comment length + extra length
+    // should not exceed 65,535 bytes per the PKWare APP.NOTE
+    // 4.4.10, 4.4.11, & 4.4.12.
+    private static final int MAX_COMBINED_CEN_HEADER_SIZE = 0xFFFF;
     /**
      * Creates a new ZIP entry with the specified name.
      *
      * @param  name
      *         The entry name
      *
      * @throws NullPointerException if the entry name is null
-     * @throws IllegalArgumentException if the entry name is longer than
-     *         0xFFFF bytes
+     * @throws IllegalArgumentException if the combined length of the entry name
+     * and {@linkplain #CENHDR CEN Header size} exceeds 65,535 bytes.
```

Comment:
English is not my native language, but should 'CEN header size" be preceded by the definite article "the" here?

Reply:
Not really needed but added anyways